File: CountBookBackend/CountBookBackend.Data/Record.cs
using System;
using System.ComponentModel.DataAnnotations;
using NodaTime;

namespace CountBookBackend.Data
{
  public class Record : IUserEntity
  {
    public int Id { get; set; }
    public RecordType Type { get; set; }

    public string Name { get; set; }

    public decimal Amount { get; set; }
    
    public int CategoryId { get; set; }
    
    public Category Category { get; set; }

    public LocalDate Date { get; set; }
    
    [Required]
    public string ApplicationUserId { get; set; }
    
    public ApplicationUser ApplicationUser { get; set; }
  }
}

File: CountBookFrontend/src/app/services/record.ts
import {DatePeriod} from "./date-period";

export interface Record {
  id: number;
  name: string;
  amount: number;
  type: RecordType;
  categoryId: number;
  date: string;
}

export enum RecordType{
  Income,
  Expense
}

export interface CreateRecordModel{
  name: string;
  amount: number;
  type: RecordType;
  categoryId: number;
  date: string;
}

export interface RecordFilterModel{
  categoryId: number;
  date: DatePeriod;
  recordType: RecordType;
}
